src/main/groovy/com/github/benmanes/gradle/versions/reporter/HtmlReporter.groovy

```diff
@@ -0,0 +1,287 @@
+      printStream.println("<table class=\"warningInfo\">")
+      getUnresolvedRows(result).each { printStream.println it }
+      printStream.println("</table>")
+      printStream.println("<br>")
+    }
+  }
+
+  private def getUnresolvedRows(Result result) {
+    List<String> rows = new ArrayList<>();
+    // The following dependencies are using the latest milestone version:
+    DependenciesGroup<DependencyUnresolved> list = result.getUnresolved()
+    rows.add("<thead><tr><th>Name</th><th>Group</th><th>URL</th><th>Current Version</th></tr></thead>")
+    for (DependencyUnresolved item : list.dependencies) {
+      String rowString;
+      String rowStringFmt = "<tr><td>%s</td><td>%s</td><td>%s</td><td>%s</td></tr>"
+      rowString = String.format(rowStringFmt, item.getName(), item.getGroup(),
+        getUrlString(item.getProjectUrl()) , getVersionString(item.getGroup(), item.getName(), item.getVersion()))
+      rows.add(rowString)
+    }
+    return rows
+  }
+
+  private def writeGradleUpdates(printStream, Result result) {
+    if (!result.gradle.isEnabled()) {
+      return
+    }
+
+    printStream.println("<H2>Gradle ${gradleReleaseChannel} updates</H2>")
+
+
+    printStream.println("Gradle ${gradleReleaseChannel} updates:")
+    result.gradle.with {
+      // Log Gradle update checking failures.
+      if (current.isFailure) {
+        printStream.println("<P>[ERROR] [release channel: ${CURRENT.id}] " + current.reason + "</P>")
+      }
+      if ((gradleReleaseChannel == RELEASE_CANDIDATE.id || gradleReleaseChannel == NIGHTLY.id) && releaseCandidate.isFailure) {
+        printStream.println("<P>[ERROR] [release channel: ${RELEASE_CANDIDATE.id}] " + releaseCandidate.reason + "</P>")
+      }
+      if (gradleReleaseChannel == NIGHTLY.id && nightly.isFailure) {
+        printStream.println("<P>[ERROR] [release channel: ${NIGHTLY.id}] " + nightly.reason + "</P>")
+      }
+
+      // print Gradle updates in breadcrumb format
+      printStream.print("<P>Gradle: [" + running.version)
+      boolean updatePrinted = false
+      if (current.isUpdateAvailable && current > running) {
+        updatePrinted = true
+        printStream.print(" -> " + current.version)
+      }
+      if ((gradleReleaseChannel == RELEASE_CANDIDATE.id || gradleReleaseChannel == NIGHTLY.id) && releaseCandidate.isUpdateAvailable && releaseCandidate > current) {
+        updatePrinted = true
+        printStream.print(" -> " + releaseCandidate.version)
+      }
+      if (gradleReleaseChannel == NIGHTLY.id && nightly.isUpdateAvailable && nightly > current) {
+        updatePrinted = true
+        printStream.print(" -> " + nightly.version)
+      }
+      if (!updatePrinted) {
+        printStream.print(": UP-TO-DATE")
+      }
+      printStream.println("]<P>")
+    }
+  }
+
+  private def getUrlString(String url) {
+    if (url == null) {
+      return "";
+    }
+    return String.format("<a target=\"_blank\" href=\"%s\">%s</a>", url, url)
+  }
```

Comment:
Could you say more on what I think are two different issues, or maybe not. I've been trying to stay as isolated as possible so venturing into other classes gives me pause. But I take direction :)

Reply:
I guess my concern is if whitespace causes you problems in html, whereas it just looks off in the other reports. The proper fix is to change Resolver to trim the the nullable string.

https://github.com/ben-manes/gradle-versions-plugin/blob/db3d3a1ea5e83a5fd04c4016bfd392c063d234fa/src/main/groovy/com/github/benmanes/gradle/versions/updates/Resolver.groovy#L337-L343

Reply:
I tested JMH by adding it to examples/gradle locally and it worked as expected. No changes required.

Reply:
okay, great. It only sticked out to me because I use that in Caffeine where it shows as,

```
 - org.openjdk.jmh:jmh-core [1.25 -> 1.25.2]

        http://openjdk.java.net/projects/code-tools/jmh/

 - org.openjdk.jol:jol-cli [0.12 -> 0.13]
     http://maven.apache.org
```

and you're right that has no effect on the html report when I tried it out. :)